lib/Site.js

```diff
@@ -81,6 +84,43 @@ function setExtension(filename, ext) {
   );
 }
 
+/**
+* Creates a function that delays invoking `func` until after `wait` milliseconds have elapsed
+* and the running `func` has resolved/rejected.
+* @param func the promise-returning function to delay,
+*        func should take in a single array
+* @param wait the number of milliseconds to delay
+* @returns delayedFunc that takes in a single argument (either an array or a single value)
+*/
+function delay(func, wait) {
```

Comment:
nit: this function should live under a utility file in `./util/` since it could be used in other places than `Site.js`.